### PR TITLE
Change default `baseBranch` in `changeset init` from `master` to `main`

### LINKED ISSUE #675: Change default base branch: master -> main
Since GitHub [changed the name of the default branch on new repos to `main`](https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/) on Oct 1, 2020, `changesets` might consider updating the default `baseBranch` to match:

```diff
$ yarn changeset init
$ cat .changeset/config.json
{
  "$schema": "https://unpkg.com/@changesets/config@1.6.2/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "linked": [],
  "access": "restricted",
- "baseBranch": "master",
+ "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "ignore": []
}
```

### PR DESCRIPTION
**Pull request by janosh**
Supersedes #677, closes #675.

### CODE REVIEW

**Comment by janosh** [comment]
Linter doesn't give information why it's failing.

File: .changeset/funny-cooks-clap.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/cli": patch
+---
+
+Change default `baseBranch` in `changeset init` from `master` to `main`.
```
Comment (Andarist) [inline .changeset/funny-cooks-clap.md:5]:
```suggestion
`changeset init` will now configure `baseBranch: "main"` in the `.changeset/config.json` as the default value.
```

**Comment by Andarist** [comment]
> Linter doesn't give information why it's failing.

It was missing a trailing new line in the file. No worries - I've fixed this and I agree that it's somewhat suboptimal that the reason behind the failure is not printed by that task.

**Comment by janosh** [comment]
> It was missing a trailing new line in the file.

I thought that might've been it but also thought it would surely say so if the problem were that simple.

**Pull request merged**